sktime/utils/dependencies/_safe_import.py

```
@@ -71,7 +72,7 @@ def _safe_import(import_path, pkg_name=None):
             return importlib.import_module(import_path)
     else:
         mock_obj = MagicMock()
-        mock_obj.__call__ = MagicMock(
-            return_value=f"Please install {pkg_name} to use this functionality."
+        mock_obj.__str__.return_value = (
```

Another question (possibly not relevant to the PR) for @fkiraly 

```pycon
>>>
>>> from unittest.mock import MagicMock
>>>
>>> a=MagicMock()
>>> a.__str__.return_value="some warning"
>>>
>>> a()
<MagicMock name='mock()' id='140147982346960'>
>>>
>>> a.b
<MagicMock name='mock.b' id='140147982212240'>
>>>
>>> a.b()
<MagicMock name='mock.b()' id='140147982204400'>
>>>
```

Since nothing fails or even warns, is it safe to use? For example, if an user (end user of sktime, or a developer creating something else using sktime) uses an estimator which uses this, at what point in the runtime will they know something is actually not going right?

---

hm, interesting question. I would say that the estimator will raise an exception in `__init__` due to missing soft dependencies. The mock object should never reach the user.